### PR TITLE
Tabs bottom border radius and separator UI

### PR DESCRIPTION
**Pull request by petemill**
Fix https://github.com/brave/brave-browser/issues/962

Brave design spec shows no bottom border radius on tabs.
This patching method, although not great, is simple compared to the amount of code which would need to be copied over and only slightly modified, since bottom border radius is calculated in the middle of a long drawing function.

Feedback appreciated.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Needed or QA/No-QA-Needed) to include the closed issue in milestone 

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by petemill** [comment]
Rebasing and testing on latest master / C70

**Comment by petemill** [comment]
This is ready for review now. @bbondy @bridiver I also tagged you for review in case you had an opinion on this new patching method (making the chromium namespace non-anonymous so that it can be accessed by the brave override class).

**Comment by petemill** [comment]
@simonhong that is valid, but it puts some of our customized logic mixed in with the chromium source, whilst we may have some other logic in the brave modules. The method currently in this PR is a proposal to keep the logic changes in the brave modules to avoid confusion. If we still prefer putting the function in the patched file over de-anonymizing the namespace, then sure - I'll make that change.

File: browser/ui/views/tabs/brave_tab.cc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  leading_separator_bounds.Offset(-origin.x(), -origin.y());
+  trailing_separator_bounds.Offset(-origin.x(), -origin.y());
+
+  const SkColor separator_base_color = controller_->GetTabSeparatorColor();
+  const auto separator_color = [separator_base_color](float opacity) {
+    return SkColorSetA(separator_base_color,
+                       gfx::Tween::IntValueBetween(opacity, SK_AlphaTRANSPARENT,
+                                                   SK_AlphaOPAQUE));
+  };
+  const int separator_radius = kSeparatorThickness;
```
Comment (simonhong) [inline browser/ui/views/tabs/brave_tab.cc]: `kSeparatorThickness/2` ?

File: browser/ui/views/tabs/brave_tab.cc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/tabs/brave_tab.h"
+
+#include "chrome/browser/ui/views/tabs/tab_controller.h"
+#include "ui/gfx/canvas.h"
+#include "ui/gfx/scoped_canvas.h"
+
+using namespace chrome_browser_ui_views_tabs_tab_cc;
+
```
Comment (simonhong) [inline browser/ui/views/tabs/brave_tab.cc]: nit: empty line

**Comment by simonhong** [comment]
@petemill , Yep, my suggestion is just another option :)

**Comment by petemill** [comment]
@simonhong addressed all feedback now - thanks. Decided to keep in the de-anonymizing namespace option after speaking to other team members due to the less worse impact on future rebasing.

File: browser/ui/views/tabs/brave_tab.cc [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                                                      separator_radius * scale);
+  cc::PaintFlags flags;
+  flags.setAntiAlias(true);
+  flags.setColor(separator_color(separator_opacities.left));
+  canvas->DrawRoundRect(
+                      leading_separator_bounds, scaled_separator_radius, flags);
+  flags.setColor(separator_color(separator_opacities.right));
+  canvas->DrawRoundRect(
+                     trailing_separator_bounds, scaled_separator_radius, flags);
+}
```
Comment (simonhong) [inline browser/ui/views/tabs/brave_tab.cc]: nit: last empty line.

File: browser/ui/views/tabs/brave_tab.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
++ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_TABS_BRAVE_TAB_H_
+#define BRAVE_BROWSER_UI_VIEWS_TABS_BRAVE_TAB_H_
+
+#include "chrome/browser/ui/views/tabs/tab.h"
+#include "ui/gfx/geometry/rect.h"
+
+namespace chrome_browser_ui_views_tabs_tab_cc {
+  const gfx::RectF ScaleAndAlignBounds(const gfx::Rect& bounds,
```
Comment (simonhong) [inline browser/ui/views/tabs/brave_tab.h]: nit: indent

File: browser/ui/views/tabs/brave_tab.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                                     float scale,
+                                     float stroke_thickness);
+
+}
+
+class BraveTab : public Tab {
+  public:
```
Comment (simonhong) [inline browser/ui/views/tabs/brave_tab.h]: nit: indent

**Comment by petemill** [comment]
0.55.x f1d811d6

**Comment by bbondy** [comment]
Added merged/0.55.x label

**Pull request merged**